src/foam/nanos/analytics/mixpanel/MixpanelService.js

```diff
@@ -108,6 +124,42 @@ foam.CLASS({
         }
         return whitelist.contains(event.getName());
       `
+    },
+    {
+      name: 'put',
+      documentation: `add message to deliveryqueue`,
+      args: 'JSONObject message',
+      javaCode: `
+        getDeliveryQueue().add(message);
+      `
+    },
+    {
+      name: 'deliver',
+      args: 'X x',
+      documentation: `batch deliver all messages in queue, called from cronjob`,
+      javaCode: `
+        var counter = getDeliveryQueue().size();
+        int messageCount = 0;
+        try {
+          ClientDelivery clientDelivery = new ClientDelivery();
+          JSONObject message = null;
+          while ( counter > 0 ) {
```

Comment:
Safer to do something like while ( ( message = getDeliveryQueue().poll() ) != null ) instead of using count.
Then it will pickup newly added entries and it won't cause issues if two threads run the deliver() method at same time. Also slightly less code.